bootstrap/index.js

```diff
@@ -1,36 +1,83 @@
-const { logger } = require('@chainlink/external-adapter')
+const { Requester, logger } = require('@chainlink/external-adapter')
+const { withCache, envOptions } = require('./lib/cache')
+const { toAsync } = require('./lib/util')
 const server = require('./lib/server')
 const gcp = require('./lib/gcp')
 const aws = require('./lib/aws')
-const { withCache, envOptions } = require('./lib/cache')
 
-const withStatusCode = (execute) => (data, callback) => {
-  // Make sure data has the same statusCode as the one sent in callback
-  const _callback = (statusCode, data) =>
-    data && typeof data === 'object' && data.statusCode
-      ? callback(statusCode, { ...data, statusCode })
-      : callback(statusCode, data)
-  return execute(data, _callback)
+const skipOnError = (middleware) => async (execute) => {
+  // Try to execute, pass through on error
+  const _safe = (executeWithMiddleware) => async (data) => {
+    try {
+      return await executeWithMiddleware(data)
+    } catch (error) {
```

Comment:
Passing the error to the next middleware sounds like functional error handling, e.g. Monads & Functors, Rx :D 

In any case, the first implementation did not skip errors but just returned a 500 to the user. When thinking about this, I realized that it is probably wrong to error the request just because it's not optimized/cached. Part of the pipeline failed, but we can still recover and fetch the data directly from the data provider.

Reply:
Makes sense for caching in particular, I'm just thinking about when we add more middlewares than just caching.

Reply:
For example, if we added an auth middleware then this would leave a pretty big bug as they would still be able to access the data provider